maven-compat/src/main/java/org/apache/maven/repository/legacy/DefaultWagonManager.java

```diff
@@ -465,6 +469,28 @@ public void getRemoteFile(
         }
     }
 
+    private void copyFile(File source, File destination) throws IOException {
+        String message;
+        if (!source.exists()) {
+            message = "File " + source + " does not exist";
+            throw new IOException(message);
+        } else if (!source.getCanonicalPath().equals(destination.getCanonicalPath())) {
+            File parentFile = destination.getParentFile();
+            if (parentFile != null && !parentFile.exists()) {
+                parentFile.mkdirs();
```

Comment:
The return value isn't checked here. Use Files.createDirectories instead and drop the exists check.

Reply:
Changing the behaviour of legacy / untested code is not a good idea.  AND maven-compat is bound to be deleted in 4.1

Reply:
The existing code is buggy and can fail. There is no guarantee mkdirs succeeds here.

Reply:
The method has been removed.